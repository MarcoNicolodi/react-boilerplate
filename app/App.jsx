import React from 'react';

const App = () => (
  <div style={{ display: 'flex', justifyContent: 'center', alignContent: 'center' }}>
    <h1>Hello World!</h1>
  </div>);

export default App;
