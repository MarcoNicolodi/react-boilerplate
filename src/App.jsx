import React from 'react';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import CssBaseline from '@material-ui/core/CssBaseline';
import MenuIcon from '@material-ui/icons/Android';

const App = () => (
  <CssBaseline>
    <div style={{ display: 'flex', justifyContent: 'center', alignContent: 'center' }}>
      <Typography variant="h1">Hello World!</Typography>
      <Button variant="contained" color="primary">
      Hello World
      </Button>
      <MenuIcon />
      <Button variant="contained" color="primary">Oi</Button>
    </div>
  </CssBaseline>
);

export default App;
