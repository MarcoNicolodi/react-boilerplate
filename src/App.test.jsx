import React from 'react';
import { shallow, mount } from 'enzyme';
import App from './App';

describe('App component', () => {
  it('should match snapshot', () => {
    const app = shallow(<App />);
    expect(app).toMatchSnapshot();
  });

  it('should display a greating to the user', () => {
    const app = mount(<App />);
    expect(app.find('h2').text()).toEqual('Make wonderful things!');
  });
});
