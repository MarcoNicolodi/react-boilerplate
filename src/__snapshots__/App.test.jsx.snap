// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Switch>
        <Route
          exact={true}
          path={
            Array [
              "/",
              "/home",
            ]
          }
          render={[Function]}
        />
        <Route
          path="/users"
          render={[Function]}
        />
        <Route
          component={[Function]}
        />
      </Switch>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            exact={true}
            path={
              Array [
                "/",
                "/home",
              ]
            }
            render={[Function]}
          />,
          <Route
            path="/users"
            render={[Function]}
          />,
          <Route
            component={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": Array [
              "/",
              "/home",
            ],
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/users",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Switch>
          <Route
            exact={true}
            path={
              Array [
                "/",
                "/home",
              ]
            }
            render={[Function]}
          />
          <Route
            path="/users"
            render={[Function]}
          />
          <Route
            component={[Function]}
          />
        </Switch>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path={
                Array [
                  "/",
                  "/home",
                ]
              }
              render={[Function]}
            />,
            <Route
              path="/users"
              render={[Function]}
            />,
            <Route
              component={[Function]}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": Array [
                "/",
                "/home",
              ],
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/users",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
